### PR TITLE
execution/new-block-tick (PR D)

### PR DESCRIPTION
**Pull request by kanej**

### CODE REVIEW

File: packages/core/src/new-api/internal/deployer.ts [hunk 1]
```diff
@@ -113,6 +115,10 @@ export class Deployer {
     const batches = Batcher.batch(module, previousStateMap);
 
     return this._executionEngine.execute({
+      block: { number: -1, hash: "-" },
```
Comment (alcuadrado) [inline packages/core/src/new-api/internal/deployer.ts:118]: I commented this on PR B, but I think everything that's passed here should be the internal state of the ExecutionEngine, and it should receive the necessary params instead.

File: packages/core/src/new-api/internal/deployer.ts [hunk 2]
```diff
@@ -30,6 +30,8 @@ import { assertIgnitionInvariant } from "./utils/assertions";
 import { getFuturesFromModule } from "./utils/get-futures-from-module";
 import { validate } from "./validation/validate";
 
+const BLOCK_POLLING_INTERVAL = 200;
```
Comment (alcuadrado) [inline packages/core/src/new-api/internal/deployer.ts:33]: Should be received by `deploy`, which means it'd be configurable by the plugin.

**Review by alcuadrado — APPROVED** [review]

File: packages/core/src/new-api/internal/deployer.ts (hunk 1 above)
Comment (kanej) [reply packages/core/src/new-api/internal/deployer.ts:118]: Agreed, I have recorded a todo for this: https://github.com/NomicFoundation/ignition/issues/311

File: packages/core/src/new-api/internal/deployer.ts (hunk 2 above)
Comment (kanej) [reply packages/core/src/new-api/internal/deployer.ts:33]: I have recorded a todo for this. It has been partially done in the timeouts PR: https://github.com/NomicFoundation/ignition/pull/306

But I have added an overall todo to check that config is being passed through the entire stack and comes from the plugin: https://github.com/NomicFoundation/ignition/issues/312

**Pull request merged**